core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2272,6 +2273,25 @@ private[spark] object Utils extends Logging {
     }.map(threadInfoToThreadStackTrace)
   }
 
+  /** Return a heap dump */
+  def getHeapDump: InputStream = {
+    try {
+      val clazz = classOf[HotSpotDiagnosticMXBean]
```

Comment:
Above you used `Class.forName()` but here you import it and load it unconditionally.
It would be safer to use `Class.forName()` again.

Reply:
Fixed